src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Logging/OutputWindowLogger.cs

```diff
@@ -25,16 +25,26 @@ internal class OutputWindowLogger : IOutputWindowLogger, IDisposable
 #endif
 
     private readonly OutputPane _outputPane;
-
+    private readonly OutputWindowLogHubLoggerProviderFactory _outputWindowLogHubLoggerProviderFactory;
     private ILogger? _testLogger;
+    private ILogger? _logHubLogger;
 
     [ImportingConstructor]
-    public OutputWindowLogger(JoinableTaskContext joinableTaskContext)
+    public OutputWindowLogger(JoinableTaskContext joinableTaskContext, OutputWindowLogHubLoggerProviderFactory outputWindowLogHubLoggerProviderFactory)
     {
         _outputPane = new OutputPane(joinableTaskContext);
+        _outputWindowLogHubLoggerProviderFactory = outputWindowLogHubLoggerProviderFactory;
+
+        _ = InitializeLogHubAsync();
+    }
+
+    private async Task InitializeLogHubAsync()
+    {
+        var loggerProvider = (LogHubLoggerProvider)await _outputWindowLogHubLoggerProviderFactory.GetOrCreateAsync("Razor.OutputWindow", CancellationToken.None);
```

Comment:
I tried to make this its own logger, and initialize it properly, but being async it meant JTF, and that just hung VS.

This is the bit that makes me think maybe I should just fix the bug, and leave logging to another time, when we can do it properly. Thoughts?

Reply:
Yea, I think I'd prefer the logging to be in a different PR. IMO, we should basically have logging that always does: import `ILoggerFactory` and create a logger. We can add different sources as needed to the factory. It's just a matter of figuring out the layering and MEF nonsense correctly. I think some of Dustin's changes to reduce MEF containers would help?

Reply:
Completely agree that would be ideal. Output window should just be another sink. I was envisaging we'd import our own IRazorLoggerFactory just because I'm paranoid about MEF, but it would be a thing wrapper. I'm convinced! Will update the PR.